### PR TITLE
V11 prep

### PR DESCRIPTION
**Pull request by did-g**
Hi,

newer versions are using TupleDescAttr for accessing tuple attributes and it has been backpatched to versions >= 9.2.

### CODE REVIEW

File: src/compat.h [hunk 1]
```diff
@@ -12,6 +12,13 @@
 #define PG96 is_supported_pg_version_96(PG_VERSION_NUM)
 #define PG10 is_supported_pg_version_10(PG_VERSION_NUM)
 
+#ifndef TupleDescAttr
+/*
+   backport in PG10 commit d34a74dd
+*/
+#define TupleDescAttr(a, i) ((a)->attrs[(i)])
+#endif
+
```
Comment (praiskup) [inline src/compat.h]: This compat thingy is IMO not needed, because TupleDescAttr was backpatched to 9.2+.

File: src/copy.c [hunk 2]
```diff
@@ -438,13 +438,15 @@ timescaledb_CopyGetAttnums(TupleDesc tupDesc, Relation rel, List *attnamelist)
 	if (attnamelist == NIL)
 	{
 		/* Generate default column list */
-		Form_pg_attribute *attr = tupDesc->attrs;
+
```
Comment (praiskup) [inline src/copy.c]: empty line

File: src/compat.h (hunk 1 above)
Comment (did-g) [reply src/compat.h]: Hi,
Yes it is backpatched to 9.2, but I was doing git bisect and pg V9.6/V10/V11 fork is older than the backpatch. An always valid TupleDescAttr during the bisect is nice to have.

Comment (praiskup) [reply src/compat.h]: I wouldn't personally expect that people build against other than reasonably new minor releases.

**Comment by davidkohn88** [comment]
Hey @did-g -Thanks for this PR! We've been working on some PG11 things ourselves and I think we're going to avoid doing it piecemeal (ie try to do all of those changes in 1 PR) so I'll be incorporating some of this into my PR if that's okay! Will of course mention you and give credit in the PR message!

**Pull request closed without merging**